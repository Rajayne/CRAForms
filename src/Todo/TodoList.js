import React from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
  return (
    <div>
      <h1>Todo List</h1>
      <Todo />
      <NewTodoForm />
    </div>
  );
};

export default TodoList;
