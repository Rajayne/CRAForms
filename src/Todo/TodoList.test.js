import { render, screen, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

test("renders BoxList without crashing", () => {
  render(<TodoList />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test("displays initialTodos", () => {
  render(<TodoList />);
  expect(screen.getByTestId(1)).toBeInTheDocument();
  expect(screen.getByTestId(2)).toBeInTheDocument();
});
