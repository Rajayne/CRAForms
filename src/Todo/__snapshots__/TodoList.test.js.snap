// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Todo List
    </h1>
    <div>
      <ul>
        <li
          data-testid="1"
          id="1"
        >
          Water the plants 
          <button>
            X
          </button>
        </li>
      </ul>
    </div>
    <div>
      <ul>
        <li
          data-testid="2"
          id="2"
        >
          Feed the dog 
          <button>
            X
          </button>
        </li>
      </ul>
    </div>
    <div>
      <h2>
        New Todo Form
      </h2>
      <form
        class="newTodoForm"
      >
        <input
          id="newTodo-input"
          name="newTodo-input"
          placeholder="New Todo"
          type="text"
          value=""
        />
        <button>
          Submit!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
